pkg/ottl/ottlfuncs/func_scale.go

```diff
@@ -0,0 +1,132 @@
+package ottlfuncs
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+type ScaleArguments[K any] struct {
+	Value      ottl.Getter[K]
+	Multiplier float64
+}
+
+func NewScaleFactory[K any]() ottl.Factory[K] {
+	return ottl.NewFactory("Scale", &ScaleArguments[K]{}, createScaleFunction[K])
+}
+
+func createScaleFunction[K any](_ ottl.FunctionContext, oArgs ottl.Arguments) (ottl.ExprFunc[K], error) {
+	args, ok := oArgs.(*ScaleArguments[K])
+
+	if !ok {
+		return nil, fmt.Errorf("ScaleFactory args must be of type *ScaleArguments[K]")
+	}
+
+	return Scale(args.Value, args.Multiplier)
+}
+
+func Scale[K any](value ottl.Getter[K], multiplier float64) (ottl.ExprFunc[K], error) {
```

Comment:
The current supported typed should work fine for the metric context, but for compatibility with the metrics transform processor, we may still want this function to work inside the [data point context](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/pkg/ottl/contexts/ottldatapoint#paths). I think at a minimum we should look at supporting the following types:
* `pmetric.ExponentialHistogramDataPoint`
* `pmetric.SummaryDataPointValueAtQuantileSlice`

Reply:
good suggestion - i have added support for `pmetric.SummaryDataPointValueAtQuantileSlice` and `pmetric.ExemplarSlice`. I have left out `pmetric.ExponentialHistogramDataPoint` for now, due to the concern I have outlined [here](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33246#discussion_r1623856230)